### PR TITLE
Adds tracking to collections artworks filters

### PR DESCRIPTION
**Pull request by ashleyjelks**
Adds analytics tracking for:
- opening filter modal
- closing filter modal
- applying filters
- clearing filters

https://artsyproduct.atlassian.net/browse/FX-1856

Screen Shot 2020-04-07 at 5 19 15 PM

Screen Shot 2020-04-07 at 5 19 27 PM

Screen Shot 2020-04-07 at 5 21 10 PM

Screen Shot 2020-04-07 at 5 19 47 PM

#skip_new_tests

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>Missing Test Files:

- `src/lib/Scenes/Collection/Helpers/__tests__/FilterArtworksHelpers-tests.ts`

If these files are supposed to not exist, please update your PR body to include "#skip_new_tests".</td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: src/lib/Components/FilterModal.tsx [hunk 1]
```diff
@@ -23,7 +33,7 @@ export const FilterModalNavigator: React.SFC<FilterModalProps> = ({ closeModal,
 
   const applyFilters = () => {
     dispatch({ type: "applyFilters" })
-    closeModal()
+    exitModal()
```
Comment (ashleyjelks) [inline src/lib/Components/FilterModal.tsx:36]: The `exitModal` and `closeModal` functions do the same thing, but for the sake of tracking they are broken into 2 different function calls as we only need tracking when a user explicitly closes the modal (either when tapping the X button or outside of the modal to close it).  `closemodal` is called for events that intentionally close the modal, and `exitModal` is called during events that close the modal as a side effect of another action (e.g. applying a filter).

File: src/lib/Components/FilterModal.tsx [hunk 2]
```diff
@@ -47,15 +57,26 @@ export const FilterModalNavigator: React.SFC<FilterModalProps> = ({ closeModal,
...
                   style={{ flex: 1 }}
                 />
                 <Box p={2}>
                   <ApplyButton
                     disabled={!isApplyButtonEnabled}
-                    onPress={applyFilters}
+                    onPress={() => {
+                      tracking.trackEvent({
+                        context_screen: Schema.ContextModules.Collection,
+                        context_screen_owner_type: Schema.ContextModules.Collection,
```
Comment (sweir27) [inline src/lib/Components/FilterModal.tsx:71]: #superminor but should this be `Schema.OwnerEntityTypes.Collection`?

File: src/lib/Scenes/Collection/Helpers/FilterArtworksHelpers.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import { FilterArray } from "lib/utils/ArtworkFiltersStore"
+
+export const filterArtworksParams = (appliedFilters: FilterArray) => {
```
Comment (sweir27) [inline src/lib/Scenes/Collection/Helpers/FilterArtworksHelpers.ts:3]: Nice factor!!

**Comment by sweir27** [comment]
This looks good to me! Had one minor comment, but other than that 👍 to go.

**Comment by ashleyjelks** [comment]
#merge_on_green

**Pull request merged**